File: Java/Armstrong.java
import java.util.*;
class Armstrong
{
    int arm(int n)
    {
        int c,s=0;
        while(n>0)
        {
            c=n%10;
            s=s+(int)(Math.pow(c,3));
            n=n/10;
        }
        return s;
    }
    void range(int l,int u)
    {
        int i;
        for(i=l;i<=u;i++)
          {
              if(i==arm(i))
                System.out.println(i+" ");
            }
        }
    public static void main()
    {
        Scanner sc=new Scanner(System.in);
        Armstrong ob=new Armstrong();
        int lo,up;
        System.out.println("Enter the lower and the upper limits");
        lo=sc.nextInt();
        up=sc.nextInt();
        ob.range(lo,up);
    }
}
    
    
              
      

File: C/Armstrong.c
#include <stdio.h>
#include <math.h>

int main() {
    int num, originalNum, remainder, result = 0, n = 0;

    printf("Enter an integer: ");
    scanf("%d", &num);

    originalNum = num;

    // Count number of digits
    while (originalNum != 0) {
        originalNum /= 10;
        n++;
    }

    originalNum = num;

    // Calculate sum of digits raised to the power n
    while (originalNum != 0) {
        remainder = originalNum % 10;
        result += pow(remainder, n);
        originalNum /= 10;
    }

    // Check if Armstrong
    if (result == num)
        printf("✅ %d is an Armstrong number.\n", num);
    else
        printf("❌ %d is not an Armstrong number.\n", num);

    return 0;
}